x-pack/plugins/maps/public/connected_components/map/mb/sort_layers.ts

```diff
@@ -128,7 +128,8 @@ export function syncLayerOrder(mbMap: MbMap, spatialFiltersLayer: ILayer, layerL
       if (!isLayerInOrder(mbMap, mapLayer, LAYER_CLASS.LABEL, beneathMbLayerId)) {
         moveMapLayer(mbMap, mbLayers, mapLayer, LAYER_CLASS.LABEL, beneathMbLayerId);
       }
-      beneathMbLayerId = getBottomMbLayerId(mbLayers, mapLayer, LAYER_CLASS.LABEL);
+      const bottomMbLayerId = getBottomMbLayerId(mbLayers, mapLayer, LAYER_CLASS.LABEL);
+      if (bottomMbLayerId) beneathMbLayerId = bottomMbLayerId;
```

Comment:
this is really subtle. Do you mind adding a comment?

e.g. 

```
When the corresponding bottom mb-layer cannot be found, - e.g. when the kibana-layer is invisible and has not synced with mapbox - retain the previous "beneath"-layer
```

Reply:
I commented the unit test to capture the subtleness of the edge case. Do you think more commenting is needed?